### PR TITLE
Remove the `linux-headers` option, instead allow for supplying original headers

### PR DESCRIPTION
**Pull request by no92**
The ABI check is expected to fail in this PR, as the master that's being compared against doesn't support external kernel headers.

### CODE REVIEW

**Comment by ArsenArsen** [comment]
ISTR that struct ifreq stuff elsewhere, and some discussion about whether it's even an ABI break.  Could you check other PRs for it?

**Comment by no92** [comment]
Added a section in the README explaining how the `linux_kernel_headers` meson option works.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -70,12 +70,27 @@ jobs:
...
           - name: Compile sysdeps
             run: |
-                meson \
+                meson setup \
                     "-Dc_args=['-fno-stack-protector', '-U__linux__']" \
                     "-Dcpp_args=['-fno-stack-protector', '-U__linux__']" \
                     "-Dbuild_tests=true" \
                     "-Db_sanitize=undefined" \
-                    --cross-file ci/${{matrix.sysdeps}}.cross-file compile-${{matrix.sysdeps}}
+                    "-Db_sanitize=undefined" \
```
Comment (Dennisbonke) [inline .github/workflows/ci.yml:91]: This is double.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
LGTM but one change

File: .github/workflows/ci.yml (hunk 1 above)
Comment (64) [inline .github/workflows/ci.yml:91]: duplicate line?

**Review by 64 — APPROVED** [review]

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**